doc/development/tutorials/extending_syntax.rst

```diff
@@ -0,0 +1,223 @@
+.. _tutorial-extending-syntax:
+
+Extending syntax with roles and directives
+==========================================
+
+Overview
+--------
+
+The syntax of both reStructuredText and MyST can be extended
+by creating new directives - for block-level elements -
+and roles - for inline elements.
```

Comment:
Let's make **directives** and **roles** bold, to introduce them as focuses of the tutorial?